### PR TITLE
Upgrate to stream9 and ruby 3.1

### PR DESCRIPTION
**Pull request by mayorova**
Related porta PR: https://github.com/3scale/porta/pull/3847

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,23 +10,26 @@ ENV PATH="./node_modules/.bin:$PATH" \
...
+RUN dnf -y module enable ruby:3.1 nodejs:18 mariadb:10.11 \
     && dnf install -y --setopt=skip_missing_names_on_install=False,tsflags=nodocs \
-        ruby-devel rubygem-rdoc rubygem-irb \
+        ruby-devel rubygem-irb \
         nodejs \
-        sudo which file shared-mime-info unzip jq git \
-        postgresql libpq-devel mysql-devel zlib-devel gd-devel libxml2-devel libxslt-devel \
-        make automake gcc gcc-c++ redhat-rpm-config \
+        sudo which file shared-mime-info jq git \
+        postgresql libpq-devel mariadb-devel mysql zlib-devel gd-devel libxml2-devel libxslt-devel \
```
Comment (akostadinov) [inline Dockerfile]: I believe we should use mysql library as in production. For now it is fine but has to be fixed. Otherwise we risk of missing some obscure issue.

File: Dockerfile [hunk 2]
```diff
@@ -37,7 +40,7 @@ RUN dnf -y module enable ruby:2.7 nodejs:16 mysql:8.0 \
 # https://issues.redhat.com/browse/RHEL-14497
 RUN mkdir /tmp/memkind \
     && cd /tmp/memkind \
-    && dnf download --source memkind \
+    && dnf download http://mirror.stream.centos.org/9-stream/AppStream/source/tree/Packages/memkind-1.11.0-2.el9.src.rpm \
```
Comment (akostadinov) [inline Dockerfile]: why is this change needed :thinking:

File: Dockerfile [hunk 3]
```diff
@@ -60,7 +63,7 @@ gpgkey=https://dl-ssl.google.com/linux/linux_signing_key.pub' \
   && driver=$(curl -s "https://googlechromelabs.github.io/chrome-for-testing/last-known-good-versions-with-downloads.json" | jq --arg majorVersion "$CHROME_VERSION" -r '.channels.Stable | select(.version | startswith($majorVersion | tostring)).downloads.chromedriver[] | select(.platform == "linux64") | .url') \
   && wget -N --progress=dot:giga "$driver" -O /tmp/chromedriver-linux64.zip \
   && unzip -j /tmp/chromedriver-linux64.zip -d /tmp \
-  && rm /tmp/chromedriver-linux64.zip \
+  && rm -f /tmp/chromedriver-linux64.zip \
```
Comment (akostadinov) [inline Dockerfile]: why was this needed :thinking: , the file should always exist, no?

File: Dockerfile (hunk 2 above)
Comment (jlledom) [reply Dockerfile]: I think this was solved in https://github.com/3scale/porta/pull/3847, so it shouldn't be needed here neither.

File: Dockerfile (hunk 3 above)
Comment (jlledom) [reply Dockerfile]: I guess it's just cleanup to not leave the tarball in the image. We need the binaries extracted in the command above, not the dist tarball.

**Review by jlledom — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (mayorova) [reply Dockerfile]: Yeah, for some reason this command didn't work for me for some time.... Then magically it started working :upside_down_face:

File: Dockerfile (hunk 3 above)
Comment (akostadinov) [reply Dockerfile]: I was asking about the `-f`. The `rm` was there since before. `-f` will cause command not to fail if deleting failed or file didn't exist.

Comment (mayorova) [reply Dockerfile]: yeah, honestly, I don't know why I added it, it could have easily been by mistake and not because it was needed.

**Review by akostadinov — APPROVED** [review]

**Comment by mayorova** [comment]
Closing in favor of https://github.com/3scale/system-builder/pull/41

When Chrome 127 issue is resolved, we will need to restore installing the latest version.
